### DISCUSSION TITLE
It would be better if Seq.zip returns ValueTuple

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/fsharp#16087] [General]
There should be no reason that after getting next item from the iterator to make one extra allocation to store the result, it should just give the result directly to consumer

make vfst vsnd to work with ValueTuple too

**Comment by vzarytovskii** [comment]
It will be a breaking change, we wont be making it

**Comment by brianrourkeboll** [comment]
Another option, @Xyncgas, depending on what you're doing, is to use `Seq.map2` directly instead:

```fsharp
let zs = (xs, ys) ||> Seq.zip |> Seq.map (fun (x, y) -> x + y)
```

```fsharp
let zs = (xs, ys) ||> Seq.map2 (fun x y -> x + y)
```

And it's easy enough to shadow the built-in `zip` if you like:

```fsharp
module Seq =
    let zip xs ys = (xs, ys) ||> Seq.map2 (fun x y -> struct (x, y))
```

**Reply by Xyncgas** [reply]
Thanks :)
